runtime/binding-grpc-kafka/src/main/java/io/aklivity/zilla/runtime/binding/grpc/kafka/internal/stream/GrpcKafkaProxyFactory.java

```diff
@@ -551,6 +574,15 @@ protected void onKafkaData(
             int length,
             OctetsFW extension)
         {
+            if (!GrpcKafkaState.replyOpening(state))
+            {
+                final ExtensionFW dataEx = extension.get(extensionRO::tryWrap);
+                final KafkaDataExFW kafkaDataEx =
+                    dataEx != null && dataEx.typeId() == kafkaTypeId ? extension.get(kafkaDataExRO::tryWrap) : null;
+
+                doGrpcBegin(traceId, authorization, 0L, kafkaDataEx.merged().fetch().headers());
```

Comment:
This code seems like it can NPE at `kafkaDataEx.merged()` given the logic above it.
Probably we want to either assume non-null, or default to empty headers when null.